plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Future.hs

```diff
@@ -74,92 +88,252 @@ The current value of the underlying asset is determined by an oracle. See note
 
 -}
 
+-- | Run the futures contract. Note that this does not make the initial 
+--   transaction (see 'writeInitialTx')
+runContract
+    :: ContractActions r
+    => Future
+    -> PubKey
+    -> PubKey
+    -> Contract r ()
+runContract f long short =
+    flip runReaderT f $ do
```

Comment:
I'm so used to seeing this that it took me a while to notice... why are we using monad transformers *and* extensible effects? Shouldn't we be writing the effects in extensible-effects land?

Reply:
Indeed, isn't this precisely why we have the residual effect `r`?